java/src/main/java/ai/rapids/cudf/AssertEmptyNulls.java

```diff
@@ -0,0 +1,18 @@
+package ai.rapids.cudf;
+
+public class AssertEmptyNulls {
+  /*
+  We have to read the system property to see if we should by-pass this assert. Ideally we should
+  just use `-da:AssertNonEmptyNulls` as per Java standard, but the reason for doing this is because
+  surefire-plugin doesn't expose anyway to disable assertions at the class level or package level
+  instead we can only disable asserts at the ClassLoader level. Therefore, in order for the tests to
+  pass we need to set a flag from the pom to know we are running unit-tests to allow non-empty nulls
+ */
+  private static final boolean ALLOW_NON_EMPTY_NULLS =
+      Boolean.getBoolean("ai.rapids.allow.nonempty.nulls");
+  public static void assertHasEmptyNulls(ColumnView cv) {
+    if (!ALLOW_NON_EMPTY_NULLS) {
+      assert !cv.hasNonEmptyNulls() : "Column has non-empty nulls";
```

Comment:
I would recommend calling `cv.hasNonEmptyNulls()` only if the input is of type STRING or LIST. This would avoid unnecessarily executing native method if `cv` is not of type STRING/LIST.

Reply:
> @razajafri Did you change this? cv.type.isNestedType is true for STRUCT and for LIST, but not for STRING. I think this is fine so long as strings will not mess up the experimental comparison code.

Nope, I didn't change this. I was assuming it will have the Strings. Structs can also have non-empty nulls so I assume we should test them too?

Reply:
Struct doesn't have non-empty nulls by itself but its children can have. So yeah you should also recursively call the check on its children too.